aws/scenarios/vm/agentinstall/install.go

```diff
@@ -0,0 +1,74 @@
+package agentinstall
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func Install(runner *command.Runner, env aws.Environment, params *Params, os os.OS) error {
+	agentCommand := &agentCommand{version: params.version}
+	os.Visit(agentCommand)
```

Comment:
@pducolin : Visitor pattern https://en.wikipedia.org/wiki/Visitor_pattern.

I have spent time to think about using a visitor or not (the other option would be to have a `GetOSType()` function and switch on the `GetOSType()`). 

The reasons, I decided to use a Visitor are:
- I expect to have more OS types (for example AWS linux or Red hat) and more features that depend on the OS type.
- Visitor is a nice way to not have everything in `os` package (It has a nice composability properties).
- If a new `OSType` is added then the user cannot forget to handle the new OS type.

Indeed, Visitor pattern is an advanced pattern and if you don't know it, it can be hard to undertand it. 
I will remove it for now and revisit this decision if there are many `switch GetOStype()` added in the code.

Reply:
Code updated.